cmd/bedMaxWig/bedMaxWig.go

```diff
@@ -14,15 +14,15 @@ import (
 
 func bedMaxWig(infile string, database string, chromsizeFile string, outfile string) {
 	var records []*bed.Bed = bed.Read(infile)
-	var data []*wig.Wig = wig.Read(database)
+	var wigData []wig.Wig = wig.Read(database)
 	var sizes []chromInfo.ChromInfo = chromInfo.ReadToSlice(chromsizeFile)
 	var outlist []*bed.Bed
 	var currentBed *bed.Bed = records[0]
 	var chromSlice []float64
 	var i int
 
 	for i = 0; i < len(sizes); i++ {
-		chromSlice = WigChromToSlice(data, sizes[i].Size, sizes[i].Name)
+		chromSlice = WigChromToSlice(wigData, sizes[i].Size, sizes[i].Name)
```

Comment:
Can we get rid of this variable now? Do we use chromSlice since we updated Values to be a []float64?

Reply:
I decided to leave this pull request focused on just getting these wig changes in (I have some other working that cant move forward until then), but I do plan to do a future pull request with bedMaxWig updates and these comment are super helpful on that endeavor!

1) I have already tried to remove this variable but ran into the problem that the rest of the code is written so that the wig and bed are matched up based on the order of the chrom.sizes file. And to adjust for this felt best to be done in another pull request. I dont think that changing records[k].Chrom == wigData.Values will work unless we always expect only one wig struct per chromosome. I am looking at a wig file I am currently working with and I do have only one wig struct per chrom (this wig was made from samToWig) but I'm not sure what is typical of wigs produced elsewhere. 
2) Craig, I dont have a good feel for wigs outside of the ones I'm using which were all produced by samToWig and are all just one struct per chrom (or chrom fragment). It's sounding like maybe the last idea you expressed about sorting would help to ensure both speed and low-mem in either circumstance?

Reply:
A lot of our code is only going to have one wig struct per chrom, but many will have more.  chromSlice is going to allow us to more easily deal with multiple wigs per chrom by being a place to hold all the data for a single chrom in one place.  It is not needed, but I think is worth the cost to make the code easier to write, review, debug.  While we don't often create more than one wig struct per chrom, many other programs will, and we should do our best to handle this, but we can save that for a later pull request.
Sorting both files and reading them simultaneously would be the fanciest way to do this, but would require a rewrite and I don't think it is worth the time investment right now.  The improvements you are making in this pull request are wonderful and will be a big help.  If needed, a smaller update would be to process only one chrom at a time from the wig file, but I think we should only do that if it is needed.

Reply:
Awesome! This sounds like a great plan. I'll add a github issue about the "not handling multiple wig structs for a single chrom" and remove the unnecessary for loop we discussed in convert.go . I will wait to hear from Riley before I delete that seemingly duplicated function.